### PR TITLE
Add a 2 second timeout to fetching eth1data

### LINKED ISSUE #5521: Handle poor performing eth1 data retrieval
A user reported block proposal failure due to a poor performing eth1 node.

```
[2020-04-19 16:29:12] ERROR validator: Failed to request block from beacon node blockSlot=14245 error=rpc error: code = Internal desc = Could not get ETH1 data: could not get block number from timestamp: Post "https://goerli.prylabs.net/": context deadline exceeded pubKey=0xa92c463e6c83
```

The recommended fix here is to set a short timeout for fetching eth1data for eth1data votes, if the short timeout is exceeded then a block should be produced with random/noop eth1data vote such that the proposer can still produce the block in time.

### PR DESCRIPTION
**Pull request by prestonvanloon**
This PR returns a random eth1data vote if the node has an issue retrieving data from eth1 in a timely manner.

Resolves #5521

### CODE REVIEW

File: beacon-chain/rpc/validator/proposer.go [hunk 1]
```diff
@@ -154,6 +155,9 @@ func (vs *Server) ProposeBlock(ctx context.Context, blk *ethpb.SignedBeaconBlock
 //  - Subtract that eth1block.number by ETH1_FOLLOW_DISTANCE.
 //  - This is the eth1block to use for the block proposal.
 func (vs *Server) eth1Data(ctx context.Context, slot uint64) (*ethpb.Eth1Data, error) {
+	ctx, cancel := context.WithTimeout(ctx, 2*time.Second)
```
Comment (0xKiwi) [inline beacon-chain/rpc/validator/proposer.go:158]: Can you move this to a global var? Rather than hard-coded into the function.

Comment (prestonvanloon) [reply beacon-chain/rpc/validator/proposer.go]: Done

**Comment by prestonvanloon** [comment]
Fixing tests now

File: beacon-chain/rpc/validator/proposer.go [hunk 2]
```diff
@@ -169,11 +175,13 @@ func (vs *Server) eth1Data(ctx context.Context, slot uint64) (*ethpb.Eth1Data, e
 	// Look up most recent block up to timestamp
 	blockNumber, err := vs.Eth1BlockFetcher.BlockNumberByTimestamp(ctx, eth1VotingPeriodStartTime)
 	if err != nil {
-		return nil, errors.Wrap(err, "could not get block number from timestamp")
+		log.WithError(err).Error("Failed to get block number from timestamp")
```
Comment (terencechain) [inline beacon-chain/rpc/validator/proposer.go:178]: We also could consider warning for these, i dont feel strongly for either, up to you!

Comment (prestonvanloon) [reply beacon-chain/rpc/validator/proposer.go:178]: Its an error that should be monitored

**Pull request merged**